test/test_resize.cpp

```diff
@@ -3326,4 +3326,81 @@ TEST_F(ResizeTest, NoncancellableSplitAndCat_CatOnlySubsetOfSplitOutputs) {
   EXPECT_FALSE(out_tensors[0].is_alias_of(in_tensor));
 }
 
+TEST_F(ResizeTest, CancellableSplitAndCat_PermuteInBetween) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = makeContigConcreteTensor({2, 3, 10});
+  TensorView* s0 = slice(in, {0, 0, 0}, {2, 3, 2});
+  TensorView* s1 = slice(in, {0, 0, 2}, {2, 3, 5});
+  TensorView* s2 = slice(in, {0, 0, 5}, {2, 3, 10});
```

Comment:
You might also want a test where the split/cat axis is permuted, to ensure that you are tracking that properly.